x-pack/plugins/lens/public/datatable_visualization/components/table_basic.tsx

```diff
@@ -346,6 +390,10 @@ export const DatatableComponent = (props: DatatableRenderProps) => {
         }}
       >
         <EuiDataGrid
+          key={
+            // we control the key when pagination is on to circumvent an EUI rendering bug
+            pagination && columns.map(({ id }) => id).join('-') + '-' + pagination.pageSize
+          }
```

Comment:
I was thinking of something more like:
```suggestion
            // we control the key when pagination is on to circumvent an EUI rendering bug
            // add a key only to force re-rendering when pagination is active
          ...pagination 
          ?  {key:columns.map(({ id }) => id).join('-') + '-' + pagination.pageSize}
          : {}
```

Reply:
Even better 👍